### PR TITLE
(PDB-2704) Log at-at exceptions

### PR DESCRIPTION
**Pull request by rbrw**
The way at-at's uses Java Executors causes any exceptions thrown by a
task to be silently swallowed, so wrap the task with a try/catch that
logs exceptions as errors.

You can demonstrate the problem by adding a throw to the
db-maintenance-tasks function and running this:

  lein test :only puppetlabs.puppetdb.acceptance.node-ttl/test-node-ttl

It should hang.  After this patch, the test should still hang, but
should also log the exception.

### CODE REVIEW

**Comment by rbrw** [comment]
If you merge this, leave the ticket open because we still need to fix extensions.

File: src/puppetlabs/puppetdb/cli/services.clj [hunk 1]
```diff
@@ -288,15 +288,19 @@
...
-            db-maintenance-tasks (fn []
-                                   (do
-                                     (when (seconds-pos? node-ttl) (auto-expire-nodes! node-ttl write-db))
-                                     (when (seconds-pos? node-purge-ttl) (purge-nodes! node-purge-ttl write-db))
-                                     (when (seconds-pos? report-ttl) (sweep-reports! report-ttl write-db))
-                                     ;; Order is important here to ensure
-                                     ;; anything referencing an env or resource
-                                     ;; param is purged first
-                                     (garbage-collect! write-db)))]
+            db-maintenance-tasks #(try
```
Comment (senior) [inline src/puppetlabs/puppetdb/cli/services.clj]: for long anonymous functions, I prefer the (fn [] ...) syntax.

File: src/puppetlabs/puppetdb/cli/services.clj [hunk 2]
```diff
@@ -288,15 +288,19 @@
...
+                                    (when (seconds-pos? node-purge-ttl)
+                                      (purge-nodes! node-purge-ttl write-db))
+                                    (when (seconds-pos? report-ttl)
+                                      (sweep-reports! report-ttl write-db))
+                                    ;; Order is important here to ensure
+                                    ;; anything referencing an env or resource
+                                    ;; param is purged first
+                                    (garbage-collect! write-db)
+                                    (catch Exception ex
+                                      (log/error ex)))]
```
Comment (senior) [inline src/puppetlabs/puppetdb/cli/services.clj]: Do we really want to swallow the exception here and not rethrow it? Today at-at doesn't do error handling properly, so the exception gets ignored, but do you expect it to always be that way?

Comment (rbrw) [inline src/puppetlabs/puppetdb/cli/services.clj]: Indeed - we do not.

File: src/puppetlabs/puppetdb/cli/services.clj (hunk 1 above)
Comment (rbrw) [inline src/puppetlabs/puppetdb/cli/services.clj]: Right - changed.

**Pull request merged**